src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -1426,23 +1426,51 @@ class LedgerMasterImp
         if (!referenceLedger || (referenceLedger->getLedgerSeq() < index))
             return ledgerHash; // Nothing we can do. No validated ledger.
 
-        // See if the hash for the ledger we need is in the reference ledger
+        // Is the hash for the ledger we need in the reference ledger?
         ledgerHash = referenceLedger->getLedgerHash (index);
-        if (ledgerHash.isZero())
+
+        // Yes
+        if (ledgerHash.isNonZero())
+            return ledgerHash;
+
+        // No. We need to get another ledger that might have it. That ledger
```

Comment:
If you can pull it off, it's good to avoid talking about the first person (I and we).

    // The hash is not in the reference ledger - get another ledger that might have it.  That ledger